test/Sanitizers/tsan-zero-sized-struct.swift

```diff
@@ -0,0 +1,31 @@
+// RUN: %target-swift-frontend -sanitize=thread -emit-ir -primary-file %s | %FileCheck %s
```

Comment:
Can we move this "IR generation" tests into `test/SILGen/tsan_instrumentation.swift` (or move the whole test) and add a runtime test here.  I minimized the following repro from another radar.  It reproduces even on my x86_64 machine:

```
// RUN: %target-run-simple-swift(-sanitize=thread)
// REQUIRES: executable_test
// REQUIRES: tsan_runtime
// UNSUPPORTED: OS=tvos

// Ensure ThreadSanitizer does not crash for zero-sized structs.

protocol MyProtocol {
    mutating func foo()
}

struct ZeroSizedStruct: MyProtocol {
    mutating func foo() { }
}

func callFoo<R: MyProtocol>(using x: inout R) {
    x.foo()
}

func callFoo<R: MyProtocol>(_ array: [Int], using x: inout R) {
    callFoo(using: &x)
}

var x = ZeroSizedStruct()
callFoo([0], using: &x)

```

It just runs the test to make sure we don't crash (no FileCheck).